From d438f64f946f7bb6966cd27e6d21ea2679d22d28 Mon Sep 17 00:00:00 2001
From: EminZhang <EminZhang@viatech.com.cn>
Date: Wed, 4 Feb 2015 10:48:23 +0800
Subject: [PATCH] Modify video-player app that some button can show normally
 now, because the buttons will disappear when redrawing colorkey before, and
 add a function to transmit a param to v4lsink,and use this param to check
 whether to redraw colorkey.

---
 src/video.c | 25 ++++++++++++++++---------
 1 file changed, 16 insertions(+), 9 deletions(-)

diff --git a/src/video.c b/src/video.c
index 7a5ff8f..2335a5c 100644
--- a/src/video.c
+++ b/src/video.c
@@ -25,6 +25,7 @@
 
 #include "owl-overlay-bin.h"
 #include "bacon-volume.h"
+static gboolean video_flag=FALSE;
 
 typedef struct {
         GtkWindow *window;
@@ -267,7 +268,7 @@ open_file_button_clicked_cb (GtkButton *button,
                  **/
                 set_has_media (data, TRUE);
                 gtk_toggle_button_set_active (data->play_pause_button, TRUE);
-
+		        video_flag=TRUE;
                 break;
         }
         default:
@@ -275,6 +276,12 @@ open_file_button_clicked_cb (GtkButton *button,
         }
 
         gtk_widget_destroy (file_chooser);
+        if(video_flag&&data->video_widget->priv->overlay)
+      	{
+     	  gboolean flag=FALSE;
+    	  gst_x_overlay_handle_events(data->video_widget->priv->overlay,flag);
+    	}
+
 }
 
 /**
@@ -416,7 +423,12 @@ main (int argc, char **argv)
                           G_CALLBACK (window_delete_event_cb),
                           NULL);
 
-        /**
+         /**
+         * Create VBox inside video widget.
+         **/
+
+	vbox = gtk_vbox_new (FALSE, 0);
+	 /**
          * Create video widget.
          **/
         data->video_widget = OWL_VIDEO_WIDGET (owl_video_widget_new ());
@@ -443,14 +455,9 @@ main (int argc, char **argv)
                           data);
 
         gtk_container_add (GTK_CONTAINER (data->window),
-                           GTK_WIDGET (data->video_widget));
+                           vbox);
 
-        /**
-         * Create VBox inside video widget.
-         **/
-        vbox = gtk_vbox_new (FALSE, 0);
-        gtk_container_set_border_width (GTK_CONTAINER (vbox), 12);
-        gtk_container_add (GTK_CONTAINER (data->video_widget), vbox);
+        gtk_container_add (GTK_BOX(vbox),GTK_WIDGET(data->video_widget));
 
         /**
          * Create right alignment in the bottom end of the VBox.
-- 
2.5.2

